File: backend/app/Models/Expense/Budget.php
<?php

namespace App\Models\Expense;

use App\Models\Trip;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Budget extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'total_budget',
        'trip_id',
        'description',
        'currency',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'total_budget' => 'decimal:2',
    ];

    /**
     * Get the trip that owns the budget.
     */
    public function trip(): BelongsTo
    {
        return $this->belongsTo(Trip::class, 'trip_id');
    }

    /**
     * Get the expenses for the budget.
     */
    public function expenses(): HasMany
    {
        return $this->hasMany(Expense::class, 'budget_id');
    }
}

File: mobile_frontend/lib/models/budget/budget.dart
import 'package:mobile_frontend/models/budget/expend.dart';
import 'package:mobile_frontend/models/budget/expense_category.dart';

class Budget {
  final int? budgetId;
  final int? tripId;
  final double? totalBudget;
  final String? currency;
  final double? totalSpent;
  final double? remainingBudget;
  final String? description;
  final double? dailyBudget;
  final List<Expense>? expenses;
  final Map<ExpenseCategory, double>? categoryLimits;

  Budget({
    this.budgetId,
    this.totalBudget,
    this.currency,
    this.expenses,
    this.dailyBudget,
    this.categoryLimits,
    this.tripId,
    this.totalSpent,
    this.remainingBudget,
    this.description,
  });
  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    if (other is! Budget) return false;
    return budgetId == other.budgetId &&
        totalBudget == other.totalBudget &&
        currency == other.currency;
  }

  @override
  int get hashCode => super.hashCode ^ budgetId.hashCode ^ totalBudget.hashCode;
}
